<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>812</width>
    <height>653</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <item>
       <widget class="QPushButton" name="connect_btn">
        <property name="text">
         <string>OCD Connect</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="ip_line_edit">
        <property name="inputMask">
         <string/>
        </property>
        <property name="placeholderText">
         <string>IP Address</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="port_line_edit">
        <property name="placeholderText">
         <string>Port No</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="title">
         <string>Commands</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="readmem_button">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Read Mem</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="hex_rdbtn">
             <property name="text">
              <string>Hex</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">rdm_bgroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="int_rdbtn">
             <property name="text">
              <string>Integer</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">rdm_bgroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="rmem_ledit">
             <property name="placeholderText">
              <string>Mem Address</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="rsize_ledit">
             <property name="placeholderText">
              <string>Size</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="data_to_wrt_btn_4">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QPushButton" name="wrtmem_button">
             <property name="text">
              <string>Write Mem</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="whex_rdbtn">
             <property name="text">
              <string>Hex</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">wrm_btngroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="wint_rdbtn">
             <property name="text">
              <string>Integer</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">wrm_btngroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="wmem_ledit">
             <property name="placeholderText">
              <string>Write Address</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="wsize_ledit">
             <property name="placeholderText">
              <string>Size</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="data_to_wrt_btn">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QPushButton" name="reset_button">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Reset</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="halt_rdbtn">
             <property name="text">
              <string>Halt</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">rst_btngroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="run_rdbtn">
             <property name="text">
              <string>Run</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">rst_btngroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QPushButton" name="flashim_button">
             <property name="text">
              <string>Flash Image</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="image_path_ledit"/>
           </item>
           <item>
            <widget class="QToolButton" name="select_elf_btn">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QPushButton" name="cstm_cmd_btn">
             <property name="text">
              <string>Command</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="cstm_cmd_ledit">
             <property name="placeholderText">
              <string>Command</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="custom_arg_ledit">
             <property name="placeholderText">
              <string>Arguments</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTextEdit" name="terminal_text">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(180, 180, 180);
color: rgb(255, 255, 255);
font: 12pt &quot;Consolas&quot;;
</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>812</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_openocd_path"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="action_openocd_path"/>
   <addaction name="action_elf_path"/>
  </widget>
  <action name="action_elf_path">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentOpen"/>
   </property>
   <property name="text">
    <string>Elf File</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="action_openocd_path">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::FolderOpen"/>
   </property>
   <property name="text">
    <string>OpenOCD </string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::EditClear"/>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="rdm_bgroup"/>
  <buttongroup name="rst_btngroup"/>
  <buttongroup name="wrm_btngroup"/>
 </buttongroups>
</ui>
